sdk/v2/meta/errors.go

```diff
@@ -50,9 +50,15 @@ type ErrNotFound struct {
 	// ID is the identifier of the resource of type Type that could not be
 	// located.
 	ID string `json:"id,omitempty"`
+	// Reason is a natural language explanation around why the resource could not
+	// be located.
+	Reason string `json:"reason,omitempty"`
 }
 
 func (e *ErrNotFound) Error() string {
+	if e.Reason != "" {
+		return e.Reason
+	}
```

Comment:
What do you think about offering the explanation in addition to the existing text as opposed to instead of?

Reply:
That was indeed my first approach, but I found the output in this specific case slightly awkward... I was going to have Type be JobLogs and ID be the job name... so then the combined error messsage reads: "JobLogs myjobname not found: ....".  Granted, this is a specific case that shouldn't preclude others... but I also envisioned that there might be other cases where Type/ID aren't available or relevant and we just want to supply the reason.

Oh, we could also do it this way: if Type, ID and Reason are non-empty, combine 'em.  if only Type, ID exist, print the standard 'not found' message. If only Reason exists, just print that.  WDYT?

Reply:
I understand the concern you're presenting and I like your proposed solution. 👍